src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationServiceRunner.cs

```diff
@@ -422,11 +444,11 @@ static bool IsCapturedPrimaryConstructorParameter(
                 if (enclosingNamedType is null)
                     return false;
 
-                if (symbol is not IParameterSymbol parameterSymbol)
-                    return false;
+                Debug.Assert(symbol is IParameterSymbol);
+                var parameterSymbol = (IParameterSymbol)symbol;
 
-                if (!parameterSymbol.IsPrimaryConstructor(cancellationToken))
-                    return false;
+                // Do not allow cancelling the assertion
```

Comment:
I don't know what this comment means

Reply:
It refers to passing default as the cancellation token in the evaluation of whether the parameter symbol is a primary constructor parameter. Though the assertion should also handle the case of the cancellation token having been cancelled, so I'll adjust to `Debug.Assert(parameter.IsPrimaryConstructor(cancellationToken) || cancellationToken.IsCancellationRequested);`

Reply:
I still don't understand. Why not just pass the cancellation token

Reply:
I thought that if the cancellation token was passed, and a cancellation was requested, the method would return false. Haven't got much knowledge about whether cancellations throw or early-exit and in which areas. If it throws, then simply passing the cancellation token will do.

Reply:
They throw